### PR TITLE
Make instantiating _qname free -- SIMICS-19564

### PR DESCRIPTION
**Pull request by lwaern-intel**

### CODE REVIEW

File: include/simics/dmllib.h [hunk 1]
```diff
@@ -1428,6 +1428,38 @@ __qname(dml_qname_cache_t *cache, const char *fmt, ...)
...
+UNUSED static const char *
+_DML_get_qname(_identity_t id, const _id_info_t *id_infos,
+               dml_qname_cache_t *cache) {
+    _id_info_t info = id_infos[id.id];
+
+    if (info.dimensions == 0) {
+        return info.logname;
+    }
+
+    uint32 *indices = MM_MALLOC(info.dimensions, uint32);
```
Comment (mandolaerik) [inline include/simics/dmllib.h:1440]: why not a VLA?

Comment (lwaern-intel) [reply include/simics/dmllib.h:1440]: eh. I was a little paranoid of `info.dimensions` could get corrupted or something. There's no real reason.
Would you prefer a VLA?

Comment (lwaern-intel) [reply include/simics/dmllib.h]: After some discussion on r.c., now a VLA.

**Review by mandolaerik — APPROVED** [review]

**Comment by syssimics** [comment]
Verification [#10367090](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=10367090): pass

**Pull request merged**